package ru.practicum.shareit.booking.mapper;

import lombok.experimental.UtilityClass;
import ru.practicum.shareit.booking.dto.BookingDto;
import ru.practicum.shareit.booking.model.Booking;
import ru.practicum.shareit.item.mapper.ItemMapper;
import ru.practicum.shareit.user.mapper.UserMapper;

@UtilityClass
public class BookingMapper {

	public BookingDto toDto(Booking booking) {
		if (booking == null) {
			return null;
		}

		return new BookingDto(
				booking.getId(),
				booking.getStart(),
				booking.getEnd(),
				ItemMapper.toDto(booking.getItem()),
				UserMapper.toDto(booking.getBooker()),
				booking.getStatus()
		);
	}
}